python_modules/dagster/dagster/_core/definitions/schedule_definition.py

```diff
@@ -745,12 +763,65 @@ def evaluate_tick(self, context: "ScheduleEvaluationContext") -> ScheduleExecuti
             skip_message = None
 
         # clone all the run requests with the required schedule tags
-        run_requests_with_schedule_tags = [
-            request.with_replaced_attrs(
-                tags=merge_dicts(request.tags, DagsterRun.tags_for_schedule(self))
+        run_requests_with_schedule_tags = []
+
+        for run_request in run_requests:
+            if run_request.partition_key and not run_request.has_resolved_partition_run_request():
+                if context.repository_def is None:
+                    raise DagsterInvariantViolationError(
+                        "Must provide repository def to build_schedule_context when yielding"
+                        " partitioned run requests"
+                    )
+
+                scheduled_target = (
+                    self.load_target()
+                    if isinstance(self._target, DirectTarget)
+                    else context.repository_def.get_job(self._target.pipeline_name)
+                )
+
+                if isinstance(scheduled_target, GraphDefinition):
+                    raise Exception(
+                        f"Error in schedule {self._name}: Schedule returned a partitioned"
+                        " RunRequest, but the target is a graph with no partitions definition."
+                    )
+
+                if isinstance(scheduled_target, PipelineDefinition):
+                    if not scheduled_target.is_job:
+                        raise Exception("Schedule target cannot be a legacy pipeline.")
+                    else:
+                        scheduled_target = cast(JobDefinition, scheduled_target)
+
+                partitions_def = scheduled_target.partitions_def
+                if not partitions_def:
+                    raise Exception(
+                        f"Error in schedule {self._name}: Schedule returned a partitioned"
+                        " RunRequest, but the target job has no partitions definition."
+                    )
+
+                dynamic_partitions_store = None
+                if isinstance(partitions_def, DynamicPartitionsDefinition):
+                    # Instance also needs to be provided when partitions def is a multipartitions
```

Comment:
Is this a TODO?

Reply:
Also, feel free to ignore, but it could make sense to push this (and maybe the check above) down into `with_resolved_tags_and_config` so this code doesn't need to be duplicated twice.

Reply:
Thinking a little more on this - it's not clear to me that we need to go out of our way to provide this nice error message, because `evaluate_tick` is not a public function - we wouldn't expect anyone to use it with `build_schedule_context`.

Reply:
ah yeah makes sense--I think i just got confused since it appears in one of our api docs examples. I'll remove it https://docs.dagster.io/_apidocs/schedules-sensors#dagster.build_schedule_context